src/test/java/org/opentripplanner/middleware/ApiKeyManagementTest.java

```diff
@@ -1,27 +1,126 @@
 package org.opentripplanner.middleware;
 
-import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
+import org.opentripplanner.middleware.models.AdminUser;
+import org.opentripplanner.middleware.models.ApiKey;
+import org.opentripplanner.middleware.models.ApiUser;
+import org.opentripplanner.middleware.persistence.PersistenceUtil;
+import org.opentripplanner.middleware.utils.HttpUtils;
+
+import java.io.IOException;
+
+import static org.opentripplanner.middleware.persistence.PersistenceUtil.getApiUser;
+import static org.opentripplanner.middleware.persistence.PersistenceUtil.getStatusCodeFromResponse;
+import static org.opentripplanner.middleware.persistence.PersistenceUtil.removeAdminUser;
+import static org.opentripplanner.middleware.persistence.PersistenceUtil.removeApiUser;
 
 /**
  * Tests for creating and deleting api keys. The following config parameters are required for these tests to run:
  *
- * DISABLE_AUTH set to true to bypass auth checks.
+ * DISABLE_AUTH set to true to bypass auth checks and use users defined here.
  * DEFAULT_USAGE_PLAN_ID set to a valid usage plan id. AWS requires this to create an api key.
- * OTP_MIDDLEWARE_URI_ROOT, e.g. http://localhost:4567.
  */
-// FIXME: User is derived from token so difficult to test. To be addressed at a later date.
 public class ApiKeyManagementTest extends OtpMiddlewareTest {
 
-    @Test @Disabled
-    public void createApiKeyForUser() {
+    private static ApiUser apiUser;
+    private static AdminUser adminUser;
+
+    /**
+     * Create an {@link ApiUser} and an {@link AdminUser} prior to unit tests
+     */
+    @BeforeAll
+    public static void setUp() throws IOException {
+        OtpMiddlewareTest.setUp();
+        apiUser = PersistenceUtil.createApiUser("test@example.com");
+        adminUser = PersistenceUtil.createAdminUser("test@example.com");
+    }
+
+    /**
+     * Remove the users {@link AdminUser} and {@link ApiUser} and any remaining API keys.
+     */
+    @AfterAll
+    public static void tearDown() {
+        removeAdminUser(adminUser.id);
+
+        // refresh api keys
+        apiUser = getApiUser(apiUser.id);
+        for (ApiKey apiKey : apiUser.apiKeys) {
+            // delete api keys if present
+            String url = String.format("http://localhost:4567/api/secure/application/%s/apikey/%s",
+                apiUser.id,
+                apiKey.id);
+            getStatusCodeFromResponse(url, HttpUtils.REQUEST_METHOD.DELETE, apiUser.auth0UserId);
+        }
+
+        removeApiUser(apiUser.id);
+    }
+
+    /**
+     * Ensure that an {@link ApiUser} can create an API key for self.
+     */
+    @Test
+    public void canCreateApiKeyForSelf() {
+        String url = String.format("http://localhost:4567/api/secure/application/%s/apikey", apiUser.id);
```

Comment:
This appears to perform a live request to the AWS API Gateway. It would be good to refactor the ApiGatewayUtils to mock the AmazonApiGateway returned in the `getAmazonApiGateway` method that can return the results we want in the test cases.